experimental/MatroidRealizationSpaces/test/runtests.jl

```diff
@@ -149,3 +149,69 @@ end
+
+    R1, mat_M1 = Oscar.selfprojecting_realization_matrix(M1, [1,2,4]); #this should be (nothing, nothing)
+    R2, mat_M2 = Oscar.selfprojecting_realization_matrix(M2, [1,2,3]);
+    R3, mat_M3 = Oscar.selfprojecting_realization_matrix(M3, [4,5,6,7]);
+    #R4, mat_M4 = Oscar.selfprojecting_realization_matrix(M4, [1,2,3,5]);#this should give an error ! How to test this?!
+    R5, mat_M5 = Oscar.selfprojecting_realization_matrix(M5, [1,5,6]);
+    R6, mat_M6 = Oscar.selfprojecting_realization_matrix(M6, [1,2,5,7]);#this should give (Ring, nothing)
+    
+    x2 = gens(R2)
+    x3 = gens(R3)
+    x5 = gens(R5)
+    x6 = gens(R6)
+
+    @testset "is_selfprojecting" begin
+        @test is_selfprojecting(M1) == true
+        @test is_selfprojecting(M2) == true
+        @test is_selfprojecting(M3) == true
+        @test is_selfprojecting(M4) == false
+        @test is_selfprojecting(M5) == true
+        @test is_selfprojecting(M6) == true
+    end
+    @testset "selfprojecting_realization_matrix" begin
+        
+        @test coefficient_ring(R2) == QQ
+        @test coefficient_ring(R3) == QQ
+        @test coefficient_ring(R5) == QQ
+        
+        @test length(x2) == 4
+        @test length(x3) == 5
+        @test length(x5) == 3
+        @test length(x6) == 3
+
+        @test mat_M1 ==  nothing
+        @test mat_M2 ==  matrix(R2, [1 0 0 1 1 1;0 1 0 1 x2[1] x2[3];0 0 1 1 x2[2] x2[4]])
+        @test mat_M3 ==  matrix(R3,  [1 1 1 1 0 0 0 0; 1 x3[1] x3[3] 0 1 0 0 0; 1 x3[2] x3[4] 0 0 1 0 1; 1 x3[2] x3[4] 0 0 0 1 x3[5]])
+        @test mat_M5 ==  matrix(R5, [1 1 1 1 0 0 1 1; 0 1 x5[1] x5[2] 1 0 0 0; 0 0 0 0 0 1 1 x5[3]])
+        @test mat_M6 == nothing
+    end
+    @testset "selfprojecting_realization_space_ideal" begin
+        #R2 and R3 are quotient rings. For the selfprojecting realization ideal we need the underlying polynomial ring
+        y2 = gens(base_ring(R2))
+        y3 = gens(base_ring(R3))
+        RR1 = base_ring(defining_ideal(realization_space(M1,char=0)))
+
+        #add a test for a completely not realizable matroid? Haven't found an example yet... 
+        @test selfprojecting_realization_ideal(M1) == ideal(R1,[1]) 
+        @test selfprojecting_realization_ideal(M2) == ideal(base_ring(R2),[y2[1]*y2[2]*y2[3] - y2[1]*y2[2]*y2[4] - y2[1]*y2[3]*y2[4] + y2[1]*y2[4] + y2[2]*y2[3]*y2[4] - y2[2]*y2[3]])
+        @test selfprojecting_realization_ideal(M3) ==ideal(base_ring(R3),[y3[1]*y3[2]*y3[3] - y3[1]*y3[2]*y3[4] - y3[1]*y3[3]*y3[4] + y3[1]*y3[4] + y3[2]*y3[3]*y3[4] - y3[2]*y3[3]])
+       # @test selfprojecting_realization_ideal(M4) #this should give an error because M4 is not selfprojecting, I don't know how to test this
```

Comment:
if you catch the error with `@req` you can use this here 

`@test_throws ArgumentError`